inlang/source-code/sdk/src/loadProject.ts

```diff
@@ -106,11 +119,7 @@ export async function loadProject(args: {
 		const setSettings = (settings: ProjectSettings): Result<void, ProjectSettingsInvalidError> => {
 			try {
 				const validatedSettings = parseSettings(settings)
-				if (validatedSettings.experimental?.persistence) {
-					settings["plugin.sdk.persistence"] = {
-						pathPattern: projectPath + "/messages.json",
-					}
-				}
+				v2Persistence = !!validatedSettings.experimental?.persistence
 
 				batch(() => {
```

Comment:
@martin-lysk could you clarify the intent of wrapping the next two calls inside a batch()?

Reply:
```
batch(() => {
	// reset the resolved modules first - since they are no longer valid at that point
	setResolvedModules(undefined)
	_setSettings(validatedSettings)
})
```

when setSettings is called with a new settings object the call to _setSettings will trigger an effect - modules that depending on settings and will load the modules asynchrouns will refelect the old state at this point.

If something like  createMessagesQuery has an effect on settings AND on resolve modules it would get triggered with an invalid state. We call _setSettings and setResolvedModules in a batch to make sure the state is reflected in the effect. 

The execution call with the batch looks like this:

1. setSettings is called and the new settings object is set and the modules are reset (not loaded yet) and the module load is triggered https://github.com/opral/monorepo/blob/ae47203e54f1e0f41d0c25bff7aafc3cf90a1cc5/inlang/source-code/sdk/src/loadProject.ts#L145
2. an effect in messageReportQuery is triggered for the settings. it will return early since the modules are not ready yet
3. the modules get loaded and an effect on resolvedModules is triggerd
4. another effect in messageReportQuery is triggered - this time with settings and resolved modules

Reply:
ok, thanks for clarifying the flow.